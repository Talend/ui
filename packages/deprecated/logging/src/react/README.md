# React error logger

This module register a React updates hook that sends a log to an endpoint.

You need to configure 2 info
* the log server url
* the app state to send

## Configuration

You need to configure the logger in order to fit your backend logging service and add your app state.

```javascript
import { store as cmfstore } from '@talend/react-cmf';
import initReactLogger from '@talend/log/lib/react';

const store = cmfstore.initialize(appReducer);
initReactLogger({
    serverUrl: 'http://localhost:8888/error',
    getState: () => store.getState(),
    processState: state => removeSensitiveData(state),
});

```

You need to remove the user data from app state.
