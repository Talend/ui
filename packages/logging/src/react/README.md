# React error logger

This module register a React updates hook that sends a log to an endpoint.

You need to configure 2 info
* the log server url
* the app state to send

## Configuration

You need to configure the logger in order to fit your backend logging service and add your app state.

```javascript
import React from 'react';
import { store as cmfstore } from '@talend/react-cmf';
import ErrorReporter from '@talend/log/lib/react';

const store = cmfstore.initialize(appReducer);

ReactDOM.render(
	<ErrorReporter
	    serverUrl={'http://localhost:8888/error'}
	    getState={store.getState}
	    processState={state => removeSensitiveData(state)}>
        <MyApp />
    </ErrorReporter>,
	document.getElementById('app')
);
```

You need to remove the user data from app state.
