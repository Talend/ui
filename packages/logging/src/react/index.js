import initReactLogger from './react-logger';

export default initReactLogger;
