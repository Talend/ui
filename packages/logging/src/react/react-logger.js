import ReactUpdates from 'react-dom/lib/ReactUpdates';
import ReactDefaultBatchingStrategy from 'react-dom/lib/ReactDefaultBatchingStrategy';
import { initErrorTransformer, TraceKit } from '../api/errorTransformer';
import getStatePayloadMiddleware from '../api/payloadMiddleware';

function injectLogger() {
	let isHandlingError = false;
	const ReactTryCatchBatchingStrategy = {
		// this is part of the BatchingStrategy API. simply pass along
		// what the default batching strategy would do.
		get isBatchingUpdates() {
			return ReactDefaultBatchingStrategy.isBatchingUpdates;
		},

		batchedUpdates(...args) {
			try {
				ReactDefaultBatchingStrategy.batchedUpdates(...args);
			} catch (e) {
				if (isHandlingError) {
					// our error handling code threw an error. just throw now
					throw e;
				}

				isHandlingError = true;
				try {
					TraceKit.report(e);
				} finally {
					isHandlingError = false;
				}
			}
		},
	};

	ReactUpdates.injection.injectBatchingStrategy(ReactTryCatchBatchingStrategy);
}

export default function initReactLogger({ serverUrl, getState, processState = state => state }) {
	initErrorTransformer(serverUrl, {
		payloadMiddleware: getStatePayloadMiddleware(() => processState(getState())),
	});
	injectLogger();
}
