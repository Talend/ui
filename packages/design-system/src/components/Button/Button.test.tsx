import { describe, it, expect } from '@jest/globals';
import { axe } from 'jest-axe';
import { render } from '@testing-library/react';
import { Button } from './';

describe('Button', () => {
	it('should render a11y html', async () => {
		const { container } = render(
			<main>
				<Button onClick={jest.fn()} variant="primary">
					Primary
				</Button>
				<Button onClick={jest.fn()} variant="destructive" icon="trash">
					Destructive
				</Button>
				<Button onClick={jest.fn()} variant="secondary" disabled>
					Secondary disabled
				</Button>
				<Button onClick={jest.fn()} variant="tertiary" type="submit">
					Tertiary submit
				</Button>
				<Button onClick={jest.fn()} variant="tertiary" isLoading>
					Tertiary isLoading
				</Button>
				<Button onClick={jest.fn()} variant="tertiary" isDropdown>
					Tertiary isDropdown
				</Button>
			</main>,
		);
		expect(container.firstChild).toMatchSnapshot();
		const results = await axe(document.body);
		expect(results).toHaveNoViolations();
	});
});
