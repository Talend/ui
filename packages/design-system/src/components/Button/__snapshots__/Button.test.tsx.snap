// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a11y html 1`] = `
<main>
  <button
    aria-busy="false"
    class="theme-clickable theme-button theme-primary"
    type="button"
  >
    <span
      class="theme-stack theme-justify-start theme-align-center theme-nowrap theme-row theme-block theme-gap-x-XS theme-gap-y-XS"
    >
      Primary
    </span>
  </button>
  <button
    aria-busy="false"
    class="theme-clickable theme-button theme-destructive"
    type="button"
  >
    <span
      class="theme-stack theme-justify-start theme-align-center theme-nowrap theme-row theme-block theme-gap-x-XS theme-gap-y-XS"
    >
      <span
        class="theme-button__icon"
      >
        <svg
          aria-hidden="true"
          pointer-events="none"
          shape-rendering="geometricPrecision"
          style="width: 1rem; height: 1rem;"
        >
          <use
            xlink:href="#trash:M"
          />
        </svg>
      </span>
      Destructive
    </span>
  </button>
  <button
    aria-busy="false"
    class="theme-clickable theme-button theme-secondary"
    disabled=""
    type="button"
  >
    <span
      class="theme-stack theme-justify-start theme-align-center theme-nowrap theme-row theme-block theme-gap-x-XS theme-gap-y-XS"
    >
      Secondary disabled
    </span>
  </button>
  <button
    aria-busy="false"
    class="theme-clickable theme-button theme-tertiary"
    type="submit"
  >
    <span
      class="theme-stack theme-justify-start theme-align-center theme-nowrap theme-row theme-block theme-gap-x-XS theme-gap-y-XS"
    >
      Tertiary submit
    </span>
  </button>
  <button
    aria-busy="true"
    class="theme-clickable theme-button theme-tertiary"
    type="button"
  >
    <span
      class="theme-stack theme-justify-start theme-align-center theme-nowrap theme-row theme-block theme-gap-x-XS theme-gap-y-XS"
    >
      <span
        class="theme-button__loading"
      >
        <svg
          aria-hidden="true"
          data-test="button.loading"
          viewBox="0 0 16 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M8 0a8 8 0 100 16A8 8 0 008 0zm0 14.227A6.227 6.227 0 118 1.773a6.227 6.227 0 010 12.454z"
              fill="currentColor"
              opacity=".2"
            />
            <path
              d="M11.11 2.611l.886-1.534A7.951 7.951 0 008 0v1.773c1.134 0 2.195.308 3.11.838z"
              fill="currentColor"
            >
              <animatetransform
                attributeName="transform"
                dur=".85s"
                from="0 8 8"
                repeatCount="indefinite"
                to="360 8 8"
                type="rotate"
              />
            </path>
          </g>
        </svg>
      </span>
      Tertiary isLoading
    </span>
  </button>
  <button
    aria-busy="false"
    class="theme-clickable theme-button theme-tertiary"
    type="button"
  >
    <span
      class="theme-stack theme-justify-start theme-align-center theme-nowrap theme-row theme-block theme-gap-x-XS theme-gap-y-XS"
    >
      Tertiary isDropdown
      <span
        class="theme-button__caret"
      >
        <svg
          aria-hidden="true"
          pointer-events="none"
          shape-rendering="geometricPrecision"
          style="width: 0.75rem; height: 0.75rem;"
        >
          <use
            xlink:href="#chevron-down:S"
          />
        </svg>
      </span>
    </span>
  </button>
</main>
`;
